src/main/kotlin/io/github/intellij/dlanguage/lexer/DHighlightingLexer.kt

```diff
@@ -1,14 +1,12 @@
 package io.github.intellij.dlanguage.lexer
 
-import com.intellij.lexer.FlexAdapter
 import com.intellij.lexer.LayeredLexer
-import io.github.intellij.dlanguage.DlangLexer
 import io.github.intellij.dlanguage.psi.DlangTypes.*
 
 /**
  * Second step highlight pass to handle errors in tokens (string contents for example)
  */
-class DHighlightingLexer : LayeredLexer(FlexAdapter(DlangLexer())) {
+class DHighlightingLexer : LayeredLexer(DlangLexer()) {
```

Comment:
is the new DlangLexer an implementation of FlexAdapter

Reply:
No, it’s a simple `LexerBase` but uses the generated Lexer (with jflex) internally. `LexerAdapter` is an adapter for JFlex lexer to a `LexerBase` (Jetbrain interface).